spec/mail/messagedelivery.mdown

```diff
@@ -0,0 +1,221 @@
+# MessageDelivery
+
+To send a message, you create a MessageDelivery object. This contains information about how to send the message.
+
+A **MessageDelivery** object has the following properties:
+
+- **id**: `String`
+  The id of the message delivery. This is server-set and immutable.
+- **messageId**: `String`
+  The id of the message to send. This is immutable.
+- **threadId**: `String`
+  The thread id of the message to send. This is immutable and set by the server based on the message.
+- **envelope**: `Envelope|null`
+  Information for use when sending via SMTP. This is immutable. An **Envelope** object has the following properties:
+
+    - **mailfrom**: `String`
+    The email address to use as the return address in the SMTP delivery. This
+    may be the empty string.
+    - **rcptTo**: `String[]`
+    An array of email addresses to send the message to.
+
+    If the *envelope* property is `null` or omitted on creation, the server MUST generate this from the referenced message as follows:
+
+    - **mailfrom**: The email in the *From* header, if present.
+    - **rcptto**: The deduplicated list of emails from the *To*, *Cc* and *Bcc*
+      headers, if present.
+
+- **sendAt**: `Date|Duration|null`
+  The date to release the message for delivery. This is immutable. The server MUST override this property to the current date-time if the value given by the client is omitted, `null`, or in the past, or if the server does not support delayed sending. If a duration is specified, the server MUST set this property to the time of this duration from "now".
+- **maybeCanUnsend**: `Boolean`
```

Comment:
IMO then **sentMaybeCancellable** better reflects this. It is the **canUnsend** that looks strange to me.